src/convertToObject.js

```diff
@@ -6,7 +6,32 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const mainObj = {};
+
+  const workArr = sourceString.split(';');
+
+  const cleanArr = workArr
+    .map(function (item) {
+      const result = item.replace(/\n/g, '').trim();
+
+      return result;
+    })
+    .filter((item) => item !== '');
+
+  for (const pair of cleanArr) {
+    const index = pair.indexOf(':');
+
+    if (index > -1) {
+      const key = pair.slice(0, index);
+      const trimKey = key.trim();
+      const value = pair.slice(index + 1);
+      const trimValue = value.trim();
+
+      mainObj[trimKey] = trimValue;
```

Comment:
Instead of using a for-of loop, consider using an array method like `forEach` to iterate over `cleanArr`. This would be more in line with the functional programming style used earlier in the function.